src/rxn_network/costs/calculators.py

```diff
@@ -0,0 +1,81 @@
+from typing import List
+
+from itertools import product
+
+import numpy as np
+
+from rxn_network.core import Calculator
+from rxn_network.reactions import ComputedReaction
+from rxn_network.thermo.chempot_diagram import ChempotDiagram
+
+from pymatgen.analysis.phase_diagram import PhaseDiagram
+
+
+class ChempotDistanceCalculator(Calculator):
+    """
+    Calculates the chemical potential distance for a reaction (in eV/atom).
+    """
+
+    def __init__(self, cpd: ChempotDiagram, distance_type="max",
+                 name="chempot_distance"):
+        self.cpd = cpd
+        self.name = name
+        self.type = distance_type
+
+        if distance_type=="max":
+            self._mu_func = max
+        elif distance_type=="mean":
+            self._mu_func = np.mean
+
+    def calculate(self, rxn: ComputedReaction) -> float:
+        """
+
+        Args:
+            rxn:
+
+        Returns:
+
+        """
+        distances = [
+            self.cpd.shortest_domain_distance(
+                combo[0].composition.reduced_formula,
+                combo[1].composition.reduced_formula,
+            )
+            for combo in product(rxn.reactant_entries, rxn.product_entries)
+        ]
+
+        distance = self._mu_func(distances)
+        return distance
+
+    def decorate(self, rxn: ComputedReaction) -> ComputedReaction:
+        """
+
+        Args:
+            rxn:
+
+        Returns:
+
+        """
+        if rxn.data:
+            data = rxn.data.copy()
+        else:
+            data = {}
+        data[self.name] = self.calculate(rxn)
+        return ComputedReaction(rxn.entries, rxn.coefficients, data=data,
```

Comment:
I really want to avoid inplace methods in general, but because this is just editing the data dictionary, it might be appropriate. What do you think? Make it in place?

Reply:
Made it in place now